cmake/HPX_CompilerFlagsTargets.cmake

```diff
@@ -21,6 +21,13 @@ target_compile_definitions(
   hpx_public_flags INTERFACE $<$<CONFIG:Debug>:HPX_DEBUG>
 )
 
+target_compile_definitions(
+  hpx_private_flags
+  INTERFACE $<$<CONFIG:MinSizeRel>:NDEBUG> hpx_private_flags
+  INTERFACE $<$<CONFIG:Release>:NDEBUG> hpx_private_flags
```

Comment:
Do we need the private flags twice?

Reply:
That's a bug, it shouldn't be there at all. Good catch!